lib/apiservers/service/restapi/handlers/vch_put.go

```diff
@@ -0,0 +1,339 @@
+// Copyright 2018 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/go-openapi/strfmt"
+
+	"github.com/vmware/vic/cmd/vic-machine/configure"
+	"github.com/vmware/vic/lib/apiservers/service/models"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/client"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/errors"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/handlers/target"
+	"github.com/vmware/vic/lib/apiservers/service/restapi/operations"
+	"github.com/vmware/vic/lib/config"
+	"github.com/vmware/vic/lib/install/data"
+	"github.com/vmware/vic/lib/install/management"
+	"github.com/vmware/vic/pkg/trace"
+	"github.com/vmware/vic/pkg/version"
+)
+
+// TODO [AngieCris]: move it to some util package
```

Comment:
Instead of moving this, I think we'll need to replace this with a more API-friendly pattern. Timeouts don't really make sense here. (This affects upgrade and rollback as well.)

Let's create a separate issue for this.

Reply:
Yes for API side timeout doesn't make any sense at all. But it is indeed a VCH configuration and the docker connectivity and appliance initialization check will cancel the context if timeout is exceeded. I don't really know a good way to do this so that the API doesn't have to worry about it. It doesn't make much sense for the install/data side to make a default timeout for the VCH either.

Reply:
We'll probably need to change the way appliance initialization works (or have another team do so) to make failures easier to identify.

Reply:
That sounds like probably a bigger issue - executor.ConfigureVCH currently includes appliance initialization check, but on API I think what we ultimately want to do is to move the initialization part to an asynchronous background process. For this background process I don't know how we should define timeout and context deadline.
We have a ticket tracking this https://github.com/vmware/vic/issues/6924, but it doesn't look like the priority at least right now (I guess)